File: Robot/src/model/enums.js
module.exports = {
  TipoLink: class TipoLink {
    tipo = {
      code: -1,
      desc: null,
    };

    static print = {
      code: 0,
      desc: 'print',
    };

    static image = {
      code: 1,
      desc: 'image',
    };

    constructor(code) {
      if (code == TipoLink.print.code) {
        this.code = TipoLink.print.code;
        this.desc = TipoLink.print.desc;
      }
      if (code == TipoLink.image.code) {
        this.code = TipoLink.image.code;
        this.desc = TipoLink.image.desc;
      }
    }
  },
  TipoProcesso: class TipoProcesso {
    code = -1;
    desc = null;

    static partida = {
      code: 0,
      desc: 'partida',
    };

    static classificacao = {
      code: 1,
      desc: 'classificacao',
    };

    static rodada = {
      code: 2,
      desc: 'rodada',
    };

    constructor(code) {
      if (code == TipoProcesso.partida.code) {
        this.code = TipoProcesso.partida.code;
        this.desc = TipoProcesso.partida.desc;
      }
      if (code == TipoProcesso.classificacao.code) {
        this.code = TipoProcesso.classificacao.code;
        this.desc = TipoProcesso.classificacao.desc;
      }
      if (code == TipoProcesso.rodada.code) {
        this.code = TipoProcesso.rodada.code;
        this.desc = TipoProcesso.rodada.desc;
      }
    }
  },
};

File: Futebox/Futebox/Models/Utils/Enums.cs
namespace Futebox.Models.Enums
{
    public enum EnumCampeonato
    {
        Indefinido = -1,
        BrasileiraoSerieA2021 = 767,
        BrasileiraoSerieB2021 = 769,
        Libertadores2021 = 771,
        Paulistao2022 = 789,
        BrasileiraoSerieA2022 = 803,
        BrasileiraoSerieB2022 = 804,
    }

    public enum OrigemDado
    {
        footstats
    }

    public enum PageViewModes
    {
        padrao,
        igv,
        yts,
        ytv,
    }

    public enum CategoriaVideo
    {
        partida,
        classificacao,
        rodada,
        jogosdia,
    }

    public enum TipoLink
    {
        print,
        image
    }

    public enum StatusProcesso
    {
        Criado = 0,
        //Agendado = 10,

        GerandoImagem = 20,
        ImagemErro = 25,
        ImagemOK = 29,

        GerandoAudio = 30,
        AudioErro = 35,
        AudioOK = 39,

        GerandoVideo = 40,
        VideoErro = 45,
        VideoOK = 49,

        Publicando = 50,
        PublicandoErro = 55,
        PublicandoOK = 59,

        Erro = 100
    }

    public enum RobotResultCommand
    {
        OK,
        ERROR,
        AUTHFAILED,
        BLANK,
        INVALID,
    }

    public enum RedeSocialFinalidade
    {
        NENHUMA,
        YoutubeShorts,
        YoutubeVideo,
        InstagramVideo
    }
}
